core/runtime/v2/shim_manager.go

```diff
@@ -106,9 +96,22 @@ func init() {
 			if !ok {
 				return nil
 			}
-			const newPluginName = string(plugins.ShimPlugin) + ".shim"
-			pluginConfigs[originalPluginName] = nil
-			pluginConfigs[newPluginName] = original
+			src := original.(map[string]interface{})
+			dest := map[string]interface{}{}
+
+			if v, ok := src["sched_core"]; ok {
+				if schedCore, ok := v.(bool); schedCore {
+					dest["env"] = []string{"SCHED_CORE=1"}
+				} else if !ok {
+					log.G(ctx).Warnf("skipping migration for non-boolean 'sched_core' value %v", v)
+				}
+
+				delete(src, "sched_core")
```

Comment:
skipping but deleting.. should we mention it is being removed from src

Reply:
The warning mentions it, its not removed from the original source though. As @ZhangShuaiyi showed, if this isn't a boolean then it will error out in previous versions. I could have it just error out instead, its just such an edge case.